modules/rosa-sts-account-roles-terraform.adoc

```diff
@@ -0,0 +1,246 @@
+    rosa_openshift_version = var.openshift_version
+    account_role_policies  = data.rhcs_policies.all_policies.account_role_policies
+    operator_role_policies = data.rhcs_policies.all_policies.operator_role_policies
+    all_versions           = data.rhcs_versions.all
+    tags                   = var.tags
+  }
+EOF
+----
+
+.. You define the account role prefix structure in the `output.tf` file. This output definition allows you to specify how the various generated roles are constructed. Run the following command to create your `output.tf` file:
++
+[source,terminal]
+----
+$ cat<<-EOF>output.tf 
+  output "account_role_prefix" {
+    value = module.create_account_roles.account_role_prefix
+  }
+EOF  
+----
+
+.. The `variables.tf` allows you to specify values you want for select variables. If you exported a variable for the `account_role_prefix` earlier, leave this variable's default value blank. Setting the variable in both places with different values can produce unexpected results. Run the following command to create your `variables.tf` file:
++
+[IMPORTANT]
+====
+Do not include your OCM token in this file if it is not stored in a safe location.
+====
++
+[source,terminal]
+----
+$ cat<<-EOF>variables.tf 
+  variable "openshift_version" {
+    type = string
+    default = "4.13"
+    description = "Enter the desired OpenShift version as X.Y. This version should match what you intend for your ROSA cluster. For example, if you plan to create a ROSA cluster using '4.13.10', then this version should be '4.13'. You can see the supported versions of OpenShift by running 'rosa list version'."
+  }
+
+  variable "account_role_prefix" {
+    type    = string
+    default = ""
+    description = "Your account roles are prepended with whatever value you enter here. The default value in the ROSA CLI is 'ManagedOpenshift-' before all of your account roles."
+  }
+
+  variable "tags" { <1>
+    type        = map
+    default     = null
+    description = "(Optional) List of AWS resource tags to apply."
+  }
+EOF
+----
++
+--
+<1> The `tags` parameter uses a map of strings variable. The format that it takes looks like the following example:
++
+[source,terraform]
+----
+variable "tags" {
+  type    = "map"
+  default = {
+    "us-east-1" = "image-1234"
+    "us-west-2" = "image-4567"
+  }
+}
+----
+--
+. In the directory where you saved these Terraform files, run the following command to set up Terraform to create these resources:
++
+[source,terminal]
+----
+$ terraform init
+----
+. Optional: Run the following command to confirm that the Terraform code you copied is correct:
++
+[source,terminal]
+----
+$ terraform validate
+----
++
+.Sample output
++
+[source,terminal]
+----
+Success! The configuration is valid.
+----
+. Optional: Test your Terraform template and create a reusable Terraform plan file by running the following command:
++
+[source,terminal]
+----
+$ terraform plan -out account-roles.tfplan -var token=<OCM_token>
```

Comment:
Should `<OCM_token>` use our `$RHCS_TOKEN` environment variable instead?

Reply:
Also, encountered [this issue](https://stackoverflow.com/questions/68169800/terraform-is-not-finding-my-passed-json-variables) when I was testing this - led to the following error whether or not I was using the environment variable above:

```
[lbailey@twinkpad ~]$ terraform plan -out account-roles.tfplan -var token=$RHCS_TOKEN
╷
│ Error: Value for undeclared variable
│ 
│ A variable named "token" was assigned on the command line, but the root module does not declare a variable of that
│ name. To use this value, add a "variable" block to the configuration.
```

Resolved by adding the following line at the top of `main.tf` file, just before the `terraform` definition:

```
variable "token" {}
```

This was with Terraform v1.6.0 on Fedora 38 for the record.

Reply:
Actually, that variable definition should probably be:

```
variable "token" {
  type = string
}
```

Just for safety.

Reply:
This is a mistake, the token shouldn't be passed as variable.
The provider itself looking for environment variable `RHCS_TOKEN` so user just need to export the environment variable
For more info see: https://registry.terraform.io/providers/terraform-redhat/rhcs/latest/docs#authentication-and-configuration